swap_meet/vendor.py

```diff
@@ -1,2 +1,68 @@
+from operator import attrgetter
+
 class Vendor:
-    pass
\ No newline at end of file
+
+    def __init__(self, inventory=None):
+        '''Initializes a Vendor object with optional inventory argument'''
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        '''Takes in item object, appends it to inventory'''
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        '''Takes in item object, removes it from inventory''' 
+        if item not in self.inventory:
+            return False    
+        self.inventory.remove(item)
+        return item
+
+    def get_by_category(self, category):
+        '''Input: category(string), returns inventory list with items of that category'''
+        return [item for item in self.inventory if item.category == category]
+        
+    def swap_items(self, other, my_item, their_item):
+        '''Inputs: other Vendor object, this vendor's item object, other vendor's item.
+        Swaps item objects between this vendor and other vendor's inventories'''
+
+        if my_item in self.inventory and their_item in other.inventory:
```

Comment:
Ooo, I see that this line does a lot of heavy lifting in the file, since many other methods call `swap_items()` (L42, L 59, L68) to return the proper boolean.

This works great just as it is, but once we move onto bigger projects and more files, we should consider the value of adding validation checks before we run functions! For example, in `swap_best_by_category()`, we might say that if my vendor does not have any items in the category the other vendor wants, we can return `False` right away and save this computation.